thunder/dynamo/splitter.py

```diff
@@ -135,6 +135,19 @@ def callback(node) -> int:
     original_split_gm: torch.fx.GraphModule = split_module(
         gm, root_m=None, split_callback=callback, keep_original_order=True, keep_original_node_name=True
     )
+
+    def add_output(m):
+        has_output = False
+        for node in m.graph.nodes:
```

Comment:
Is it possible to iterate over nodes in reverse order? Should the last node be always the "output" node?

Reply:
`reversed(m.graph.nodes)` gives the reverse order, I'm not sure but normally it should be the last node. 
Do we need to iterate all the `call_module` and check if the output exists(like the `call_module` inside another `call_module`) or only check `original_split_gm.children()` is enough?
cc: @kshitij12345

Reply:
I think `original_split_gm.children` should suffice as we know that the bug is in `split_module` (and so checking only the partitions created by it should be good).